core/domain/classifier_domain_test.py

```diff
@@ -41,4 +42,73 @@ def test_to_dict(self):
             expected_classifier_dict['cached_classifier_data'],
             expected_classifier_dict['data_schema_version'])
         self.assertDictEqual(expected_classifier_dict,
-                             observed_classifer.to_dict())
+                             observed_classifier.to_dict())
+
+    def test_validation(self):
+        classifier_dict = {
+            'classifier_id': 1,
+            'exp_id': 'exp_id1',
+            'exp_version_when_created': 1,
+            'state_name': 'a state name',
+            'algorithm_id': "LDAStringClassifier",
+            'cached_classifier_data': {'alpha': 1.0},
+            'data_schema_version': 1
+        }
+        classifier = classifier_domain.Classifier(
+            classifier_dict['classifier_id'],
+            classifier_dict['exp_id'],
+            classifier_dict['exp_version_when_created'],
+            classifier_dict['state_name'],
+            classifier_dict['algorithm_id'],
+            classifier_dict['cached_classifier_data'],
+            classifier_dict['data_schema_version'])
+        with self.assertRaises(utils.ValidationError):
+            classifier.validate()
```

Comment:
Can you add a newline after each "subtest block", and add a brief comment above each block to explain what that block is testing? Otherwise it's hard for someone looking at the code to parse this.